python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,373 @@
+GRANT_TYPE = "grant_type"
+SCOPE = "scope"
+TOKEN_EXCHANGE = "urn:ietf:params:oauth:grant-type:token-exchange"
+
+NAMESPACE_SEPARATOR = b"\x1F".decode("UTF-8")
+
+
+class RestTable(Table):
+    config: Properties = Field(default_factory=dict)
+
+
+class CreateTableRequest(IcebergBaseModel):
+    name: str = Field()
+    location: Optional[str] = Field()
+    table_schema: Schema = Field(alias="schema")
+    partition_spec: Optional[PartitionSpec] = Field(alias="partition-spec")
+    write_order: Optional[SortOrder] = Field(alias="write-order")
+    stage_create: bool = Field(alias="stage-create", default=False)
+    properties: Properties = Field(default_factory=dict)
+
+
+class TokenResponse(IcebergBaseModel):
+    access_token: str = Field()
+    token_type: str = Field()
+    expires_in: int = Field()
+    issued_token_type: str = Field()
+
+
+class ConfigResponse(IcebergBaseModel):
+    defaults: Properties = Field()
+    overrides: Properties = Field()
+
+
+class ListNamespaceResponse(IcebergBaseModel):
+    namespaces: List[Identifier] = Field()
+
+
+class NamespaceResponse(IcebergBaseModel):
+    namespace: Identifier = Field()
+    properties: Properties = Field()
+
+
+class UpdateNamespacePropertiesResponse(IcebergBaseModel):
+    removed: List[str] = Field()
+    updated: List[str] = Field()
+    missing: List[str] = Field()
+
+
+class ListTableResponseEntry(IcebergBaseModel):
+    name: str = Field()
+    namespace: Identifier = Field()
+
+
+class ListTablesResponse(IcebergBaseModel):
+    identifiers: List[ListTableResponseEntry] = Field()
+
+
+class ErrorResponseMessage(IcebergBaseModel):
+    message: str = Field()
+    type: str = Field()
+    code: int = Field()
+
+
+class ErrorResponse(IcebergBaseModel):
+    error: ErrorResponseMessage = Field()
+
+
+class RestCatalog(Catalog):
+    token: TokenResponse
+    config: ConfigResponse
+
+    host: str
+
+    def __init__(self, name: str, properties: Properties, host: str, token: str):
+        self.host = host
+        self.token = self._fetch_access_token(token)
+        self.config = self._fetch_config()
+        super().__init__(name, properties)
+
+    def _split_token(self, token: str) -> Tuple[str, str]:
+        """Splits the token in a client id and secret
+
+        Args:
+            token: The token with a semicolon as a separator
+
+        Returns:
+            The client id and secret
+        """
+        client, secret = token.split(":")
+        return client, secret
+
+    @property
+    def headers(self) -> Properties:
+        return {AUTHORIZATION_HEADER: f"{BEARER_PREFIX} {self.token.access_token}", "Content-type": "application/json"}
+
+    def url(self, endpoint: str, prefixed: bool = True, **kwargs) -> str:
+        """Constructs the endpoint
+
+        Args:
+            prefixed: If the prefix return by the config needs to be appended
+
+        Returns:
+            The base url of the rest catalog
+        """
+
+        url = self.host
```

Comment:
It looks like `host` doesn't correctly describe the use of this setting, since it must include `v1`. I think that Python should use the same config settings that are used in most of the Java library, which to set the catalog URI using `uri`. And to match Java, that should not include `v1/`, which should be added to the path in this method.

Reply:
Good one! I assumed that it was part of the host, but it shouldn't

Reply:
Can you rename this to `uri` to match the standard catalog config options?